src/erlcloud_aws.erl

```diff
@@ -264,23 +264,106 @@ format_timestamp({{Yr, Mo, Da}, {H, M, S}}) ->
       io_lib:format("~4.10.0b-~2.10.0b-~2.10.0bT~2.10.0b:~2.10.0b:~2.10.0bZ",
                     [Yr, Mo, Da, H, M, S])).
 
+%%%---------------------------------------------------------------------------
+-spec default_config() -> aws_config().
+%%%---------------------------------------------------------------------------
+%% @doc Generate a default config
+%%
+%% This function will generate a default configuration, using credentials
+%% available in the environment, if available.  If no credentials are
+%% available then this will just return a default <code>#aws_config{}</code>
+%% record.
+%%
 default_config() ->
     case get(aws_config) of
         undefined -> default_config_env();
         Config -> Config
     end.
 
 default_config_env() ->
-    case {os:getenv("AWS_ACCESS_KEY_ID"), os:getenv("AWS_SECRET_ACCESS_KEY")} of
-        {KeyId, Secret} when is_list(KeyId), is_list(Secret) ->
-            #aws_config{access_key_id = KeyId, secret_access_key = Secret};
-        _ -> default_config_profile()
+    case config_env() of
+        {ok, Config} -> Config;
+        {error, _} -> #aws_config{}
+    end.
+
+
+%%%---------------------------------------------------------------------------
+-spec auto_config() -> {ok, aws_config()} | undefined.
```

Comment:
Hi Paul,

What about returning an actual error instead of `undefined` for `auto_config()` because metadata request to AWS may fail due to various reasons?